avocado/utils/archive.py

```diff
@@ -125,12 +125,12 @@ def __init__(self, filename, mode='r'):
         self.filename = filename
         self.mode = mode
         engine = None
-        for ext in ArchiveFile._extension_table:
+        for ext, value in ArchiveFile._extension_table:
             if filename.endswith(ext):
                 (self.is_zip,
                  self.is_tar,
                  engine,
-                 extra_mode) = ArchiveFile._extension_table[ext]
+                 extra_mode) = value
```

Comment:
Unless I'm missing something, `ArchiveFile._extension_table` is a dict, and this will raise a `ValueError` unless you use `.items()` on line 128:

```python
In [9]: for ext, value in ArchiveFile._extension_table: 
   ...:     print(value) 
   ...:                                                                                                                                                                                                               
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-6b2437ed1443> in <module>
----> 1 for ext, value in ArchiveFile._extension_table:
      2     print(value)
      3 

ValueError: too many values to unpack (expected 2)
```

Did I missed something?